vault/request_handling.go

```diff
@@ -393,6 +393,10 @@ func (c *Core) wrapInCubbyhole(req *logical.Request, resp *logical.Response) (*l
 	resp.WrapInfo.Token = te.ID
 	resp.WrapInfo.CreationTime = creationTime
 
+	if resp.Auth != nil {
```

Comment:
A comment here would help understand that `Auth` will be non-nil only when a token is created and hence that is the only case when this will be set.